serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-openapi-based-services.adoc

```diff
@@ -0,0 +1,358 @@
+:dotnet_swagger_url: https://docs.microsoft.com/en-us/aspnet/core/tutorials/web-api-help-pages-using-swagger?view=aspnetcore-6.0
+:npm_swagger_url: https://www.npmjs.com/package/swagger-ui-express
+:php_swagger_url: https://github.com/zircote/swagger-php
+:quarkus_config_url: https://quarkus.io/guides/config-reference
+:quarkus_rest_client_url: https://quarkus.io/guides/rest-client
+:mp_config_env_vars_url: https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#environment-variables-mapping-rules
+// Referenced documentation pages.
+:getting-familiar-with-our-tooling: xref:getting-started/getting-familiar-with-our-tooling.adoc
+:create-your-first-workflow-service: xref:getting-started/create-your-first-workflow-service.adoc
+:build-workflow-image-with-quarkus-cli: xref:cloud/build-workflow-image-with-quarkus-cli.adoc
+:understanding-jq-expressions: xref:core/understanding-jq-expressions.adoc
+:configuring-openapi-services-endpoints: xref:service-orchestration/configuring-openapi-services-endpoints.adoc
+:camel-k-integration: xref:integrations/camel-k-integration.adoc
+:authention-support-for-openapi-services: xref:security/authention-support-for-openapi-services.adoc
+
+This document describes how to call REST services via an link:{open_api_spec_url}[OpenAPI] specification file.
+
+The examples in this document assume that you have the Quarkus tooling installed and have created a Serverless Workflow project. 
+
+For more information about the tooling, see {getting-familiar-with-our-tooling}[Using Serverless Workflow tooling]. To create a new project, see {create-your-first-workflow-service}[Creating your first Serverless Workflow service].
+
+== The OpenAPI Function Definition
+
+Kogito follows the link:{spec_doc_url}#using-functions-for-restful-service-invocations[CNCF Serverless Workflow specification function definition] to make calls to remote REST services using an OpenAPI specification reference, as shown in the following example:
+
+.An example of an OpenAPI function definition
+[source,json]
+----
+{
+   "functions":[
+      {
+         "name":"myFunction1",
+         "operation":"classpath:/myopenapi-file.yaml#myFunction1"
+      }
+   ]
+}
+----
+
+The `operation` attribute is a string composed by:
+
+1. The URI that the engine uses to locate the specification file
+2. The operation identifier (`operationId`), which you find in the link:{open_api_spec_url}#fixed-fields-7[OpenAPI specification file].
+
+Kogito supports the following URI schemes:
+
+* `classpath` for files located in the application project under the `src/main/resources` folder. The `classpath` is the default scheme when you don't define a scheme in the string. For example `classpath:/myopenapifile.yaml` or `/myopenapifile.yaml`. In this example, the file location is in `src/main/resources/myopenapifile.yaml` path
+* `file` for files located in the filesystem
+* `http` or `https` for remotely located files
```

Comment:
```suggestion
* `http` or `https` for remotely located files.
```

Reply:
I'd instead not use periods at the end of bullet lists sentences: https://www.instructionalsolutions.com/blog/bulleted-list-punctuation

There're a few contraversions. I prefer to wait for Heena's review.